File: investpulse/frontend/src/types/stock.ts
export interface StockPrice {
  stock_code: string
  name: string
  price: number
  change: number
  change_pct: number
  volume: number
  market_cap: number
  high: number
  low: number
  open: number
  prev_close: number
}

export interface StockSearchResult {
  stock_code: string
  name: string
}

export interface StockCandle {
  timestamp: string
  open: number
  high: number
  low: number
  close: number
  volume: number
}

export interface KISStatus {
  configured: boolean
  message: string
}

File: investpulse/backend/app/schemas/stock.py
"""주식 관련 Pydantic 스키마."""

from pydantic import BaseModel


class StockPriceResponse(BaseModel):
    stock_code: str
    name: str
    price: int
    change: int
    change_pct: float
    volume: int
    market_cap: int
    high: int
    low: int
    open: int
    prev_close: int


class StockSearchResult(BaseModel):
    stock_code: str
    name: str


class StockCandleResponse(BaseModel):
    timestamp: str
    open: int
    high: int
    low: int
    close: int
    volume: int
